Tools/refcounts/mypy.ini

```diff
@@ -0,0 +1,10 @@
+[mypy]
+files = Tools/refcounts/lint.py
+pretty = True
+show_traceback = True
+python_version = 3.12
+
+strict = True
+warn_unreachable = True
+enable_error_code = all
```

Comment:
That's not a valid mypy configuration -- there's no `all` error code. Mypy also finds an error on this file with this configuration:

```
~/dev/cpython (tools/refcounts/lint-127443)⚡ % uvx mypy --config-file Tools/refcounts/mypy.ini
Installed 3 packages in 36ms
Tools/refcounts/mypy.ini: [mypy]: enable_error_code: Invalid error code(s): all
Tools/refcounts/lint.py:24: error: Missing type parameters for generic type "Match"  [type-arg]
    MATCH_TODO: Callable[[str], re.Match | None]
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 1 error in 1 file (checked 1 source file)
```

Maybe add this config file to https://github.com/python/cpython/blob/main/.github/workflows/mypy.yml so that we check this code with mypy in CI?

Reply:
Wait. How come it didn't detect it locally? I'll update this tomorrow!

Reply:
I think I only used `mypy Tools/refcounts/` but it might happen that I had some cached data. Anyway, I'll fix this tomorrow (I'm no more on my dev session).

Reply:
Mypy might not actually pick up the `mypy.ini` file unless it's in your CWD or you pass `--config-file <path/to/mypy.ini>`. I can't remember what the exact rules are for where it searches for config files; I might be wrong there. Anyway, if you pass `--config-file` you can't go wrong! (That's what we do in CI)